### PR TITLE
fix(leftnavtree): style updates

### LINKED ISSUE #1507: Nav divider is wrong color in dark theme
Screenshot 2024-09-05 at 7 36 53 AM

Should be gray 20 I believe (you can see [Carbon v10 website](https://v10.carbondesignsystem.com/) for correct color)

Also in light theme, we should use white background for the left nav for contrast with the gray-10 content background.

```[tasklist]
### Tasks
- [ ] Update nav divider color in dark theme
- [ ] Re-add missing nav border
- [ ] Adjust margin (saw some extra margins — probably to accommodate the banner on the demo)
- [ ] Fix theme light mode nav theme colors
```

### PR DESCRIPTION
**Pull request by szinta**
Closes #1507 

Fixes style issues mentioned in #1507 

Now following the below Color codes for dark and light themes

Dark theme:
```
Nav background:  #161616
Nav right Border: #393939;
Divider color: #393939
```

Light theme:
```
Nav background:  #ffffff
Nav right Border: #e0e0e0;
Divider color: #c6c6c6
```

### CODE REVIEW

File: packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss [hunk 1]
```diff
@@ -7,17 +7,19 @@
 
 .container {
   height: 100%;
+  width: 100%;
+  position: absolute;
   margin: spacing.$spacing-10 0;
-  background-color: theme.$background;
+  background-color: theme.$text-inverse;
```
Comment (elycheea) [inline packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss:13]:
```suggestion
  background-color: theme.$layer-01;
```

Should probably be using a layer token. I think this is it?

File: packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss
Comment (elycheea) [inline packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss:1]: I think we’re still missing the `border-right` here. (Does gatsby-theme-carbon use logical properties? Should it be `border-inline-end`?)

https://github.com/carbon-design-system/carbon-for-products-website/pull/98/files#diff-ccb5c6ab1b4e9c6892cba9f2a904889b49d1e52f336bd2d558ff07ec9ffc5aa2R17

File: packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss (hunk 1 above)
Comment (elycheea) [reply packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss:13]: @szinta brought up that `$layer` doesn’t map quite right for the themes — we need `white` in light theme but `#161616` in dark theme, which only `$text-inverse` seems to match.

She will follow the approach from regular left nav and add custom styles here.

File: packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss
Comment (szinta) [reply packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss:1]: gatsby-theme-carbon doesn't use logical properties, hence added border using `border-right` prop

File: packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss (hunk 1 above)
Comment (szinta) [reply packages/gatsby-theme-carbon/src/components/LeftNav/LeftNavTree.module.scss]: used $background token for dark theme and added custom style overrides for g10 theme

File: packages/example/src/data/nav-items.yaml [hunk 2]
```diff
@@ -48,6 +48,7 @@
       path: /guides/v3-migration
     - title: v4.0 Migration
       path: /guides/v4-migration
+  hasDivider: true
```
Comment (alisonjoseph) [inline packages/example/src/data/nav-items.yaml:51]: Was this added on purpose?

**Review by alisonjoseph — COMMENTED** [review]
this LGTM other than the question about the extra nav divider

File: packages/example/src/data/nav-items.yaml (hunk 2 above)
Comment (szinta) [reply packages/example/src/data/nav-items.yaml]: I added it for testing purpose only. I have now removed it.

**Review by alisonjoseph — APPROVED** [review]

**Review by elycheea — APPROVED** [review]
🎉

**Pull request merged**